lrlex/src/lib/parser.rs

```diff
@@ -15,8 +15,9 @@ lazy_static! {
         Regex::new(r"^%[sS][a-zA-Z0-9]*$").unwrap();
     static ref RE_EXCLUSIVE_START_STATE_DECLARATION: Regex =
         Regex::new(r"^%[xX][a-zA-Z0-9]*$").unwrap();
-    static ref RE_LEX_ESC: Regex =
-        Regex::new(r#"^([:xdigit:]|[digit]|[\\|a|b|f|n|r|t|v])"#).unwrap();
+    // In addition to traditional lex \123 \x1234, accept the \u1234 and \U12345678
+    static ref RE_LEX_ESC_LITERAL: Regex =
+        Regex::new(r"^(([xuU][[:xdigit:]])|[[:digit:]]|[abfnrtv\\])").unwrap();
```

Comment:
So I guess that there are 2 things I should mention here,

1. the `uU` in the `xdigit` part is I believe currently just passed through to regex (e.g. in the previous release), I believe i've used it in a project but have to look/update it.  As an extension beyond lex.
2. there is also `\p`, `\P`, `\d`, `\D` https://docs.rs/regex/1.6.0/regex/#syntax the new code will convert to `p`, `P`, `d`, `D`.

We probably should make a decision on whether these `regex` crate additional escape sequences that currently get passed through, but will no longer will without explicit code to allow them.

Unfortunately I don't believe that there is anything in the `regex-syntax` crate which can test these in the way that `is_meta_character` can test for regex escaping.

Reply:
I added another patch, which just allows all the escape sequences which it would previously implicitly supported through regex and documents the differences.  If regex changes in the future to add escape sequences though we would have to add them manually before they can be used.

https://github.com/softdevteam/grmtools/pull/333/commits/52d78b9da56cd531ee69a9ee99588cf0ddfe83b7